modules/mobile-agents/windows/jax-rs/src/main/java/org/wso2/carbon/mdm/mobileservices/windows/operations/util/OperationReply.java

```diff
@@ -93,44 +121,80 @@ private void generateBody() throws WindowsOperationException {
+                            new Status(HEADER_COMMAND_ID, sourceHeader.getMsgID(), HEADER_STATUS_ID,
+                                    HEADER_COMMAND_TEXT, sourceHeader.getSource().getLocURI(),
+                                    String.valueOf(Constants.SyncMLResponseCodes.AUTHENTICATION_ACCEPTED));
+                    status.add(headerStatus);
+                }
+            }
+        }
+        if (sourceSyncmlBody.getResults() != null) {
+            int ResultCommandId = ++HEADER_COMMAND_ID;
+            Status resultStatus = new Status(ResultCommandId,
+                    sourceHeader.getMsgID(),
+                    sourceSyncmlBody.getResults().getCommandId(),
+                    RESULTS_COMMAND_TEXT, null,
+                    String.valueOf(
+                            Constants.SyncMLResponseCodes.ACCEPTED));
+            status.add(resultStatus);
+
+        }
         if (sourceSyncmlBody.getAlert() != null) {
-            Status alertStatus = new Status(sourceSyncmlBody.getAlert().getCommandId(),
-                    HEADER_COMMAND_REFERENCE_ID,
+            int alertCommandId = ++HEADER_COMMAND_ID;
+            Status alertStatus = new Status(alertCommandId,
+                    sourceHeader.getMsgID(),
                     sourceSyncmlBody.getAlert().getCommandId(),
                     ALERT_COMMAND_TEXT, null,
                     String.valueOf(Constants.SyncMLResponseCodes.ACCEPTED));
             status.add(alertStatus);
+
         }
         if (sourceSyncmlBody.getReplace() != null) {
-            Status replaceStatus = new Status(sourceSyncmlBody.getReplace().getCommandId(),
-                    HEADER_COMMAND_REFERENCE_ID,
-                    sourceSyncmlBody.getReplace().getCommandId(),
-                    REPLACE_COMMAND_TEXT, null,
-                    String.valueOf(
-                            Constants.SyncMLResponseCodes.ACCEPTED)
-            );
-            status.add(replaceStatus);
+            int replaceCommandId = ++HEADER_COMMAND_ID;
+                Status replaceStatus = new Status(replaceCommandId,
+                        sourceHeader.getMsgID(),
+                        sourceSyncmlBody.getReplace().getCommandId(),
+                        REPLACE_COMMAND_TEXT, null,
+                        String.valueOf(
+                                Constants.SyncMLResponseCodes.ACCEPTED)
+                );
+                status.add(replaceStatus);
+
         }
         if (sourceSyncmlBody.getExec() != null) {
-            Status replaceStatus = new Status(sourceSyncmlBody.getExec().getCommandId(),
-                    HEADER_COMMAND_REFERENCE_ID,
-                    sourceSyncmlBody.getExec().getCommandId(),
-                    GET_COMMAND_TEXT, null,
-                    String.valueOf(
-                            Constants.SyncMLResponseCodes.ACCEPTED)
-            );
-            status.add(replaceStatus);
+            for(int z = 0; z<sourceSyncmlBody.getExec().size(); z++) {
```

Comment:
Fix formatting issues. Also, try to use an iterator based for-loop, which is considered much efficient.